File: redis/jruby/update/redis_update.rb
#! /usr/bin/jruby
# -*- coding: utf-8 -*-
#
#	redis_update.rb
#					Apr/18/2013
#
# --------------------------------------------------------
require 'rubygems'
require 'socket'
require 'json'
require 'date'
#
# --------------------------------------------------------
puts "*** 開始 ***"
#
key_in = ARGV[0]
population_in = ARGV[1].to_i
#
puts key_in,population_in
#
HOST = 'host_dbase'
PORT = 6379
ss = TCPSocket::open(HOST,PORT)
#
command = "get " + key_in + "\r\n"
ss.puts(command)
rec_0 = ss.gets
puts rec_0
if (rec_0[0 .. 2] != "$-1") then
	json_str = ss.gets
	puts json_str
	unit_aa = JSON.parse(json_str)
	unit_aa['population'] = population_in
	unit_aa['date_mod'] = Date.today
	json_str_new=JSON(unit_aa)
	print json_str_new
	command = "set " + key_in + " " + json_str_new +  "\r\n"
	ss.puts(command)
	rec_0 = ss.gets
end
#
ss.close
#
puts "*** 終了 ***"

File: redis/jython/update/redis_update.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	redis_update.py
#
#					Apr/12/2013
#
import	sys
reload(sys)
sys.setdefaultencoding('utf-8')
import	string
import	json
#
import redis.clients.jedis.Jedis
#
from time import localtime,strftime
# ------------------------------------------------------------
print ("*** 開始 ***")
#
key_in = sys.argv[1]
population_in = string.atoi (sys.argv[2])
print ("%s\t%d" % (key_in, population_in))
#
jedis = redis.clients.jedis.Jedis("host_dbase")
#
str_json = str (jedis.get(key_in))
print (str_json)
#
unit_aa = json.loads (str_json)
print ("%s\t%s\t%d\t%s" %  (key_in,unit_aa['name'], \
		unit_aa['population'],unit_aa['date_mod']))
#
unit_aa['population'] = population_in
today = strftime ("%Y-%m-%d",localtime ())
unit_aa['date_mod'] = today
json_str_new = json.dumps (unit_aa)
print (json_str_new)
jedis.set (key_in,json_str_new)
#
print ("*** 終了 ***")
#
# ------------------------------------------------------------
